src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs

```diff
@@ -63,6 +64,13 @@ public MSBuildForwardingAppWithoutLogging(IEnumerable<string> argsToForward, str
             }
         }
 
+        private bool ShallMSBuildServerBeUSed()
+        {
+            // We want to use MSBuild server only in inner loop development.
+            // We use Console.IsOutputRedirected as simple heuristics detection of possible non-inner-loop usage (Azure CI/CD, GitHub Actions, ...)
+            return UseMSBuildServerEnvVarConfig && !Console.IsOutputRedirected;
```

Comment:
I'm not a fan of this. If something is clearly broken, and it happens to be related to MSBuild server (not necessarily a bug in the server but perhaps some caching issue), then one fairly natural next step is to redirect the output to a file and look at that file to see what's up. Except that with this change, it might suddenly start working when you output to a file. Then, as a user, I think it was just an ephemeral bug, and I go back to building without redirecting output, and the error pops up again. I think you can see how that would be very confusing.

Reply:
This is very common for applications with an interactive mode. If you as a user want to explicitly dump the ANSI-code-aware output, then you usually have to provide an explicit flag (like `--interactive`). I spoke with @rokonec about also adding a check for this flag (which is already present in the SDK codebase) as an additional explicit signal from the user. The overall heirarchy would be

* explicit user enable via `--interactive` -> use interactive mode
* explicit user disable via `--interactive false` -> do not use interactive mode
* no explicit user signal -> infer based on env var/console output/etc

Reply:
@Forgind best way is to address such issues is to use /bl. However if there is such behavior it is a bug which needs to be fixed or weird/wrong custom tasks which ought to be fixed as well. So far we are not aware of any such thing, 
We hope people will report this and we will either help them to fix their buggy tasks, give them workaround, and/or fix internal bug.